DNN Platform/Library/Services/FileSystem/FolderManager.cs

```diff
@@ -838,23 +862,41 @@ public virtual IFolderInfo MoveFolder(IFolderInfo folder, string newFolderPath)
             return this.MoveFolder(folder, parentFolder);
         }
 
+        /// <summary>
+        /// Checks if a given folder path is valid.
+        /// </summary>
+        /// <param name="folderPath">The folder path.</param>
+        /// <returns>A value indicating whether the folder path is valid.</returns>
         internal virtual bool IsValidFolderPath(string folderPath)
         {
             var illegalInFolderPath = new Regex(string.Format("[{0}]", Regex.Escape(new string(Path.GetInvalidPathChars()))), RegexOptions.Compiled);
             return !illegalInFolderPath.IsMatch(folderPath) && !folderPath.TrimEnd('/', '\\').EndsWith(".");
         }
 
+        /// <summary>
+        /// Adds a log entry.
+        /// </summary>
+        /// <param name="folder">The folder to log about.</param>
+        /// <param name="eventLogType">The type of the log entry.</param>
         internal virtual void AddLogEntry(IFolderInfo folder, EventLogController.EventLogType eventLogType)
         {
-            EventLogController.Instance.AddLog(folder, PortalController.Instance.GetCurrentPortalSettings(), this.GetCurrentUserId(), string.Empty, eventLogType);
+            EventLogController.Instance.AddLog(folder, (PortalSettings)PortalController.Instance.GetCurrentSettings(), this.GetCurrentUserId(), string.Empty, eventLogType);
```

Comment:
Do we need to add an `IPortalSettings` overload for `AddLog`?

Reply:
Well, yeah, I guess, now I was not sure how we should properly do it once there, do we have IPortalSettings for DI and if so do we inject it in the EventLogController. That was the quick solution to just fix the StyleCop warnings, but I agree we should not have to cast it. Any advice ?